synapse/handlers/message.py

```diff
@@ -956,7 +1009,11 @@ async def handle_new_client_event(
             logger.exception("Failed to encode content: %r", event.content)
             raise
 
-        await self.action_generator.handle_push_actions_for_event(event, context)
+        # TODO: Skip actions for historical messages or figure out how to
+        # generate proper context.current_state_ids and state_groups for our historical events
+        # which have prev_events that reference non-persisted events because
+        # we are persisting in reverse-chronolical
+        # await self.action_generator.handle_push_actions_for_event(event, context)
```

Comment:
See TODO, another problem when persisting because the `prev_events` don't exist yet.

Reply:
Fixed by skipping when we see the historical field